include/fiction/algorithms/simulation/sidb/operational_domain.hpp

```diff
@@ -470,25 +490,39 @@ class operational_domain_impl
      */
     [[nodiscard]] operational_domain<parameter_point, operational_status> flood_fill(const std::size_t samples) noexcept
     {
+        assert((num_dimensions == 2 || num_dimensions == 3) &&
+               "Flood fill is only supported for two and three dimensions");
+
         mockturtle::stopwatch stop{stats.time_total};
 
         const auto step_point_samples = generate_random_step_points(samples);
 
-        // for each sample point in parallel
-        std::for_each(FICTION_EXECUTION_POLICY_PAR_UNSEQ step_point_samples.cbegin(), step_point_samples.cend(),
-                      [this](const auto& sp) { is_step_point_operational(sp); });
+        simulate_operational_status_in_parallel(step_point_samples);
 
-        // a queue of (x, y) dimension step points to be evaluated
+        // a queue of (x, y[, z]) dimension step points to be evaluated
```

Comment:
```suggestion
        // a queue of (x, y, z) dimension step points to be evaluated
```

Reply:
I would like to keep this as is, indicating that `z` is an optional dimension.